File: Daily/sbgo/Leetcode896-单调数列.go
package sbgo

func isMonotonic(A []int) bool {
	n := len(A)
	inc, dec := true, true
	for i := 0; i < n-1; i++ {
		if A[i] > A[i+1] {
			inc = false
		}
		if A[i] < A[i+1] {
			dec = false
		}
	}
	return inc || dec
}

File: Leetcode/PY/Leetcode896-单调数列.py
from typing import List
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        n = len(A)
        inc,dec = True,True
        for i in range(n-1):
            if A[i] > A[i + 1]:
                inc = False
            if A[i] < A[i + 1]:
                dec = False
        return dec or inc